nemo/collections/nlp/modules/common/megatron/megatron_perceiver_encoders.py

```diff
@@ -0,0 +1,271 @@
+        bias_gelu_fusion=True,
+        bias_dropout_add_fusion=True,
+        masked_softmax_fusion=True,
+        persist_layer_norm=False,
+        openai_gelu=False,
+        onnx_safe=False,
+        activation='gelu',
+        bias=True,
+        normalization='layernorm',
+        transformer_block_type='pre_ln',
+        headscale=False,
+        parent_model_type=ModelType.encoder_or_decoder,
+        hidden_steps=32,
+        num_self_attention_per_cross_attention=1,
+    ):
+        super(MegatronPerceiverEncoderModule, self).__init__()
+
+        self.pre_process = pre_process
+        self.post_process = post_process
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.init_method = init_method
+        self.model_attn_mask_type = encoder_attn_mask_type
+        self.hidden_dropout = hidden_dropout
+        self.output_layer_init_method = output_layer_init_method
+        self.parent_model_type = parent_model_type
+        self.normalization = normalization
+        self.transformer_block_type = transformer_block_type
+        self.hidden_steps = hidden_steps
+        self.num_self_attention_per_cross_attention = num_self_attention_per_cross_attention
+        self.num_attention_heads = num_attention_heads
+        self.apply_query_key_layer_scaling = apply_query_key_layer_scaling
+        self.kv_channels = kv_channels
+        self.ffn_hidden_size = ffn_hidden_size
+        self.precision = precision
+        self.fp32_residual_connection = fp32_residual_connection
+        self.activations_checkpoint_method = activations_checkpoint_method
+        self.activations_checkpoint_num_layers = activations_checkpoint_num_layers
+        self.layernorm_epsilon = layernorm_epsilon
+        self.bias_gelu_fusion = bias_gelu_fusion
+        self.bias_dropout_add_fusion = bias_dropout_add_fusion
+        self.masked_softmax_fusion = masked_softmax_fusion
+        self.persist_layer_norm = persist_layer_norm
+        self.openai_gelu = openai_gelu
+        self.onnx_safe = onnx_safe
+        self.activation = activation
+        self.bias = bias
+        self.relative_attention_num_buckets = relative_attention_num_buckets
+        self.relative_attention_max_distance = relative_attention_max_distance
+        self.headscale = headscale
+        self.hidden_dropout = hidden_dropout
+        self.attention_dropout = attention_dropout
+        self.position_embedding_type = position_embedding_type
+        self.use_cpu_initialization = use_cpu_initialization
+        self.normalization = normalization
+        self.parent_model_type = parent_model_type
+        self.transformer_block_type = transformer_block_type
+
+        assert self.num_self_attention_per_cross_attention >= 1
+        assert self.hidden_steps >= 1
+
+        self.init_hidden = torch.nn.Parameter(torch.nn.init.xavier_normal_(torch.empty(hidden_steps, hidden_size)))
+        self.init_cross_att = self._build_cross_attn_layer()
```

Comment:
No need for this anymore

Reply:
To maintain backward compatibility with models that we've been training so far, I'll let "0" be a valid argument and also the default?

Reply:
Actually, looks like I had this layer, but it was never used in the forward for old models. I'll just get rid of this altogether.